Plugins/GRPCProtobufGeneratorCommand/Plugin.swift

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import PackagePlugin
+
+@main
+struct GRPCProtobufGeneratorCommandPlugin: CommandPlugin {
+  /// Perform command, the entry-point when using a Package manifest.
+  func performCommand(context: PluginContext, arguments: [String]) async throws {
+    var argExtractor = ArgumentExtractor(arguments)
+
+    if CommandConfig.helpRequested(argumentExtractor: &argExtractor) {
+      OptionsAndFlags.printHelp(requested: true)
+      return
+    }
+
+    // MARK: Configuration
+    let commandConfig: CommandConfig
+    let inputFiles: [String]
+    do {
+      (commandConfig, inputFiles) = try CommandConfig.parse(
+        argumentExtractor: &argExtractor,
+        pluginWorkDirectory: context.pluginWorkDirectoryURL
+      )
+    } catch {
+      OptionsAndFlags.printHelp(requested: false)
```

Comment:
Is it normal to print out help if there's a parsing error? Wondering whether the error gets lost with the wall of help text.

Reply:
I don't know if it's normal but I've definitely seen it. I'm happy to remove it if you'd prefer.

Reply:
I think I'd prefer it was removed so that the diagnostics/errors are more obvious.